python_modules/dagster/dagster/_core/pipes/utils.py

```diff
@@ -236,44 +237,31 @@ def no_messages_debug_text(self) -> str:
 WAIT_FOR_LOGS_TIMEOUT = 60
 
 
-class PipesBlobStoreMessageReader(PipesMessageReader):
-    """Message reader that reads a sequence of message chunks written by an external process into a
-    blob store such as S3, Azure blob storage, or GCS.
-
-    The reader maintains a counter, starting at 1, that is synchronized with a message writer in
-    some pipes process. The reader starts a thread that periodically attempts to read a chunk
-    indexed by the counter at some location expected to be written by the pipes process. The chunk
-    should be a file with each line corresponding to a JSON-encoded pipes message. When a chunk is
-    successfully read, the messages are processed and the counter is incremented. The
-    :py:class:`PipesBlobStoreMessageWriter` on the other end is expected to similarly increment a
-    counter (starting from 1) on successful write, keeping counters on the read and write end in
-    sync.
-
-    If `log_readers` is passed, the message reader will start the passed log readers when the
-    `opened` message is received from the external process.
+TCursor = TypeVar("TCursor")
 
-    Args:
-        interval (float): interval in seconds between attempts to download a chunk
-        log_readers (Optional[Sequence[PipesLogReader]]): A set of readers for logs.
-    """
 
+class PipesThreadedMessageReader(PipesMessageReader):
     interval: float
-    counter: int
-    log_readers: Sequence["PipesLogReader"]
+    log_readers: Dict[str, "PipesLogReader"]
     opened_payload: Optional[PipesOpenedData]
 
     def __init__(
         self,
         interval: float = 10,
-        log_readers: Optional[Sequence["PipesLogReader"]] = None,
+        log_readers: Optional[Mapping[str, "PipesLogReader"]] = None,
     ):
         self.interval = interval
-        self.counter = 1
-        self.log_readers = check.opt_sequence_param(
-            log_readers, "log_readers", of_type=PipesLogReader
+        self.log_readers = (
+            {name: reader for name, reader in log_readers.items()} if log_readers else {}
```

Comment:
this can just be `log_readers or {}`

Reply:
Because `self.log_readers` must be a `Dict` (so we can assign to keys later on), while the constructor argument accepts a `Mapping` (a more general type).

Reply:
Aren't the new keys assigned in `extra_log_readers`

Reply:
They are initially assigned there but eventually get moved to `self.log_readers` once started. So we can have them all in one place. 

I'm up to changing this detail, it's not really important, but that's the current implementation.